osidb/serializer.py

```diff
@@ -806,3 +806,43 @@ class Meta:
             "groups",
             "profile",
         ]
+
+
+class FlawCommentSerializer(
```

Comment:
There will be 3 comment-related serializers (3rd added in `76a8581033d95a8a77a598c876457e2b162a158f` purely for correct schema generation). `CommentSerializer` has a docstring that says what it is for. `CommentSerializer` and `FlawCommentSerializer` are each used for very different purpose. IMO deduplicating them will result in a serializer that will be too complicated or behaving in unintended ways. Apart from that, this question is for @osoukup and @Elkasitu , not me, to answer.

Reply:
The naming may be confusing but otherwise if you need a different/extended behavior for the comment endpoint (like various mixins) the approach seems valid to me.

Reply:
You can do a number of things here, but I think at most you only need two serializers:

- One for the `/flaws` view, which is essentially always read-only
- One for the `/flaws/<id>/comments` view which allows both read and write operations

The first one is probably OK as-is.
The second one just needs to have the `flaw`, `order` and `external_system_id` fields declared as read-only attributes, and I would even go as far as having `order` and `flaw` omitted because their value is either obvious or uninteresting to clients.

You can probably also simplify the new serializer to inherit from the existing CommentSerializer so that it fits your needs with minimal code duplication, but I leave that as a wishlist item as I don't think it's such a big deal.